### PR TITLE
Chore / React hooks ESLint plugin / easy fixes

### PR DESCRIPTION
**Pull request by styrelius**
### What?

Second step towards using the `eslint-plugin-react-hooks`, in order for us to use hooks the way they say you should 😇

1) Fix linting warnings in components where fixing them wasn't complicated and where it was easy to manually test the number of re-renders etc. Easy fixes means adding dependencies to `useEffect` dependency array (without causing a lot of new re-renders), or moving some logic inside of `useEffect`. 

<br>

_Referenced ticket [here](https://hedvig.atlassian.net/browse/HVG-248)_

### CODE REVIEW

File: src/client/utils/hooks/useDocumentScroll.tsx [hunk 1]
```diff
@@ -8,25 +8,26 @@ export const useDocumentScroll = (
   }) => void,
 ) => {
   const [, setScrollTop] = React.useState(0)
```
Comment (gustaveen) [inline src/client/utils/hooks/useDocumentScroll.tsx:10]: Unrelated to this, but why isn't there a state variable here? 🙃

File: src/client/pages/Embark/index.tsx [hunk 2]
```diff
@@ -121,7 +121,16 @@ const Embark: React.FunctionComponent<EmbarkProps> = (props) => {
...
-  }, [currentPassage.id])
+  }, [
+    currentPassage.id,
+    currentPassage.url,
+    currentPassage.api,
+    currentPassage.externalRedirect,
+    history,
+    state,
+    props.baseUrl,
+    props.name,
```
Comment (gustaveen) [inline src/client/pages/Embark/index.tsx:132]: Feel free to omit in this PR, would be nice to deconstruct these 🤗

**Review by gustaveen — APPROVED** [review]
Nicely done! @styrelius 🥳

**Comment by styrelius** [comment]
Hello! Please note that I've removed changes made to `PerilModal.tsx` since those changes was a bit messy due to destructuring props. These changes will instead be in a separate PR!

File: src/client/utils/hooks/useDocumentScroll.tsx (hunk 1 above)
Comment (styrelius) [reply src/client/utils/hooks/useDocumentScroll.tsx:10]: Haha, I would also like to know that!

File: src/client/pages/Embark/index.tsx (hunk 2 above)
Comment (styrelius) [reply src/client/pages/Embark/index.tsx:132]: I agree with you but made the decision to not have those kind of changes in this PR, I think it makes it cleaner and easier to review if this is purely fixing linting stuff 😇

Comment (gustaveen) [reply src/client/pages/Embark/index.tsx:132]: Word! 👌

**Pull request merged**